### PR TITLE
protobuf: add static libs

### PR DESCRIPTION
**Pull request by jeroen**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

In the transition to cmake we lost the static libraries for arm bottles.

Some precedents:
 - https://github.com/Homebrew/homebrew-core/pull/97239
 - https://github.com/Homebrew/homebrew-core/pull/91420
 - https://github.com/Homebrew/homebrew-core/pull/68464

### CODE REVIEW

**Comment by jeroen** [comment]
The non-arm64 seem to time-out.

File: Formula/protobuf.rb
Comment (carlocab) [inline Formula/protobuf.rb:67]:
```suggestion
        site_packages = prefix/Language::Python.site_packages(python)
        with_env(LDFLAGS: "-Wl,-rpath,#{rpath(source: site_packages)} #{ENV.ldflags}") do
          system python, *Language::Python.setup_install_args(prefix, python), "--cpp_implementation"
        end
```
To silence the warnings about files with missing rpaths.

**Comment by jeroen** [comment]
Thanks for the review, committed your suggestions. Will probably still timeout on the dependent checks though.

**Comment by carlocab** [comment]
We can skip dependents here so CI will finish. I'll leave it to another maintainer for approval.

**Review by iMichka — APPROVED** [review]

**Pull request closed without merging**
